### ISSUE TITLE
BUG: "Could not find a machine at the current cursor."

### ISSUE THREAD

**Issue by danawoodman** [statelyai/xstate-tools#325]
Using the latest extension in an existing project with the following machine:

```ts
import { createMachine } from 'xstate'

const toggleMachine = createMachine({
	id: 'toggle',
	initial: 'initialState',
	states: {
		initialState: {},
	},
})

```

Always results in a `Could not find a machine at the current cursor.` error every time I try to open the XState visualizer. 

Have restarted and reloaded VSCode (which is up to date).

- MacBook Air M2
- Ventura 13.2.1

Note that the extension worked a few days ago, but no idea what is causing it to fail to start

**Comment by jer-0** [comment]
For me it's not working in any `.svelte` file.

**Comment by davidkpiano** [comment]
> For me it's not working in any `.svelte` file. or is this not supported?

I don't believe it's supported in `.svelte` files yet. Have you tried defining the machine in a different `.ts` file as a workaround?

**Comment by timkraut** [comment]
The same issue occurs in `.vue` files. Once I move the `export const machine = createMachine(...)` into a separate `.ts` file, it works as expected

**Comment by pietrofxq** [comment]
Having the same issue on windows 11. Created a file `machine.ts` with the same code as the first post and getting the same error. It would be useful if there the extension added a panel in VSCode Output to give more information about the error

**Comment by jtvisona** [comment]
Same here.

**Comment by davidkpiano** [comment]
We are currently investigating this (cc. @Andarist)

**Comment by joakimHjerpe** [comment]
I get the same error when trying to open from the Command Palette.
However, it works fine when opening using the command button 'Open Visual Editor' inserted into the code window on the line above the createMachine line.
VS Code 1.75.1
Xstate VSCode v1.14.3
Windows 10 
Thanks for the great XState library and visualizer!

**Comment by Andarist** [comment]
Currently, only TS/JS files are supported by this feature. I'd love to take a look at some repro case that involves those kind of files. Is anybody here able to provide one? I already tried a bunch of things and couldn't repro this on my own.

**Comment by joakimHjerpe** [comment]
The example code above from danawoodman on Apr 18th, gives  "Could not find a machine at the current cursor. " when I select 'XState: Open Visual Editor' in the  Command Palette.
However, with the command button 'Open Visual Editor' inserted into the code window on the line above the createMachine line it works great!

**Comment by pgularski** [comment]
I get this error too with my Angular code and its Services code. The XState plugin doesn't seem to like decorators. So if I have an `@Injectable` service, the plugin refuses to open the editor.

So having this:

`@Injectable({ providedIn: 'root' })`
`export class SomeService...`

Prevents the plugin from working.

If I comment the decorator out, it helps:

`// @Injectable({ providedIn: 'root' })`
`export class SomeService...`

Then I have to remember to uncomment it which is somewhat mundane.
Hopefully someone finds it helpful.

**Comment by pietrofxq** [comment]
Confirmed it also works for me when clicking on Open Inspector that appears above the createMachine line

**Comment by ced-mos** [comment]
I confirm that the error is still there for me with `VS Code 1.84.1` and `XState VSCode 2.0.0` on Windows when trying to open it via the command palette. However, as @pietrofxq mentioned it is working for me with opening via Open Inspector which appears above the createMachine line. But I still would love that it would work as well via the command palette to have a better DX.

Therefore, is there any update, when this could be solved?

**Comment by DaveHudson** [comment]
I had this issue too, unable to open via command palette or open inspector above createMachine. 

For me it seems like a VSCode extension conflict was the issue. After uninstalling a Mermaid extension I also had installed (https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid) and reloading VSCode the XState Editor opened fine from the command palette.

**Comment by kim-jaedeok** [comment]
i think this is because it is not clear where the cursor should be placed.
make sure the blinking part of the cursor is below the code that calls createMachine.
<img width="265" alt="Screenshot 2024-05-31 at 11 35 39 PM" src="https://github.com/statelyai/xstate-tools/assets/89368498/415f2ba4-12d2-482d-b321-a1bf8ba45072">

**Comment by MichalBryxi** [comment]
Just bumped onto the same issue and curiously it is somehow influenced by the path of my project:

When my project was at path: `/Users/michal/hefr.ch/ASS 4.2` it was not working and I was getting the `Could not find a machine at the current cursor.` error. The same with `/Users/michal/hefr.ch/ASS 42`, but when I removed the space from the path: `/Users/michal/hefr.ch/ASS-4.2` it started working (also reverted removing the dot to isolate the case).

So maybe the extension can't work with spaces in project path?

- OSX: 15.1.1 (24B91)
- vscode: 1.96.0

**Comment by DjWarmonger** [comment]
Ahh, here you go - even with v5 setup(): 

<img width="1071" height="486" alt="Image" src="https://github.com/user-attachments/assets/2b3f03ec-383c-4e1a-aacf-527b9f2b032e" />
